src/coreclr/vm/nativeimage.cpp

```diff
@@ -143,59 +143,69 @@ NativeImage *NativeImage::Open(
 
     NewHolder<PEImageLayout> peLoadedImage;
 
-    EX_TRY
+    BundleFileLocation bundleFileLocation = Bundle::ProbeAppBundle(fullPath, /*pathIsBundleRelative */ true);
+    if (bundleFileLocation.IsValid())
```

Comment:
It looks like the bundler also records the file type of everything in the bundle. Is there any reason why we would want to add a new "R2R composite" file type?

It doesn't really look like we'll need it, as long as we assume that the bundle-relative path is constructed properly at the beginning.

Reply:
I do not think we have concrete reasons to distinguish and propagate the distinction for composite r2r files vs. regular r2r assemblies. The main difference is that composite r2r file may contain native code for more than one assembly, which is opaque to the host/bundle. Also it does not seem to contain any IL (I am not sure if it actually can, but it would make no difference either).

As things are now, a separate classification for composite r2r at bundle level would not add much benefit. I think we can wait until/if we have a use case for that.